src/components/PostsList/PostsList.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, useEffect } from 'react';
+import './PostsList.scss';
+import { getAllPosts, getPosts } from '../../api/posts';
+import { Post } from '../../types';
+
+type Props = {
+  userSelectedId: string;
+  selectPost: (arg?: Post) => void;
+  post: Post | undefined;
+};
+
+export const PostsList: React.FC<Props> = ({
+                                             userSelectedId,
+                                             selectPost,
+                                             post,
+                                           }) => {
+  const [currentPostList, setPostList] = useState<Post[]>([]);
+
+  const allPosts = async () => {
+    const result = await getAllPosts();
+
+    setPostList(result);
+  };
+
+  const findposts = async () => {
+    const result = await getPosts(userSelectedId);
+
+    setPostList(result);
+  };
+
+  useEffect(() => {
+    findposts();
+  }, [userSelectedId]);
+
+  useEffect(() => {
+    allPosts();
+  }, []);
```

Comment:
The logic of getting posts by user or getting all posts could be improved by combining both functions. Currently, if the selected ID is "0", getPosts(userSelectedId) is still being called, which could cause unexpected side effects.

```suggestion

  useEffect(() => {
  const fetchPosts = async () => {
    if (userSelectedId === "0") {
      const result = await getAllPosts();
      setPostList(result);
    } else {
      const result = await getPosts(userSelectedId);
      setPostList(result);
    }
  };

  fetchPosts();
}, [userSelectedId]);

```